### PR TITLE
Some enum class conversions

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2/Editor.h [hunk 1]
```diff
@@ -31,16 +31,17 @@ namespace Editor
...
-    EDITOR_STEP_OBJECTIVE_SELECTION,    // 4
-    EDITOR_STEP_SAVE_SCENARIO,          // 5
-    EDITOR_STEP_ROLLERCOASTER_DESIGNER, // 6
-    EDITOR_STEP_TRACK_DESIGNS_MANAGER   // 7
+    ObjectSelection,       // 0
+    LandscapeEditor,       // 1
+    InventionsListSetUp,   // 2
+    OptionsSelection,      // 3
+    ObjectiveSelection,    // 4
+    SaveScenario,          // 5
```
Comment (tupaschoal) [inline src/openrct2/Editor.h:41]: Weirdly this one is never directly referenced

File: src/openrct2/scenario/Scenario.h [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 using random_engine_t = Random::Rct2::Engine;
 
+enum class Rct2EditorStep : uint8_t;
```
Comment (duncanspumpkin) [inline src/openrct2/scenario/Scenario.h:30]: RCT2

File: src/openrct2-ui/input/KeyboardShortcut.cpp [hunk 3]
```diff
@@ -98,7 +98,7 @@ static void ShortcutCloseAllFloatingWindow()
 {
     if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR))
         window_close_all();
-    else if (gS6Info.editor_step == EDITOR_STEP_LANDSCAPE_EDITOR)
+    else if (gS6Info.editor_step == RCT2EditorStep::LandscapeEditor)
```
Comment (Gymnasiast) [inline src/openrct2-ui/input/KeyboardShortcut.cpp:101]: Why call this `RCT2EditorStep`? We only put this prefix if we use a different approach than RCT2.

Comment (tupaschoal) [reply src/openrct2-ui/input/KeyboardShortcut.cpp:101]: I just reused what was already there. What do you suggest, just `EditorStep`?

Comment (Gymnasiast) [reply src/openrct2-ui/input/KeyboardShortcut.cpp:101]: Yup, just `EditorStep`.

**Comment by tupaschoal** [comment]
Done, I've also solved the merge conflict

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Could you change `EditStep` to `EditorStep`?

File: src/openrct2/ride/Ride.cpp [hunk 4]
```diff
@@ -1984,7 +1984,7 @@ void Ride::UpdateAll()
     // Remove all rides if scenario editor
     if (gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR)
     {
-        if (gS6Info.editor_step <= EDITOR_STEP_INVENTIONS_LIST_SET_UP)
+        if (EnumValue(gS6Info.editor_step) <= EnumValue(EditStep::InventionsListSetUp))
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1987]: Can you create an issue to change the type of `gS6Info.editor_step`?

Comment (duncanspumpkin) [reply src/openrct2/ride/Ride.cpp:1987]: Shouldn't this be a switch with all of the relevant steps.

Comment (tupaschoal) [reply src/openrct2/ride/Ride.cpp:1987]: I can convert it to a switch no problem.

O have already changed the type to the enuk one, haven't I? Not sure I get you Gym

File: src/openrct2/scenario/Scenario.h [hunk 5]
```diff
@@ -51,7 +53,7 @@ assert_struct_size(rct_s6_header, 0x20);
  */
 struct rct_s6_info
 {
-    uint8_t editor_step;
+    EditorStep editor_step;
```
Comment (tupaschoal) [inline src/openrct2/scenario/Scenario.h:56]: I had already changed this @Gymnasiast , is that what you meant?

File: src/openrct2/ride/Ride.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Ride.cpp]: I was thrown off by the use of `EnumValue`, but I now see why you had to use that.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**